src/lib/createCustomElement.js

```
@@ -0,0 +1,35 @@
+// export default function createCustomElement (elementName) {
+//   return function () {
+//     Array.from(document.querySelectorAll('section'))
+//         .forEach(section => section.style.display = 'none')
+//
+//     const element = document.createElement(elementName)
+//     document.body.appendChild(element)
+//   }
+// }
+
+export default function createCustomElement (object) {
+  return function () {
+    Array.from(document.querySelector('section').children)
+      .forEach(child => child.remove())
+
+    class CustomElement extends HTMLElement {
+      // Invoked by the browser when the element is instantiated
+      // by `document.createElement`.
+      createdCallback () {
+        this.innerHTML = object.content
+      }
+    }
+
+    try {
+      document.registerElement(object.element, CustomElement)
+    } catch (e) {
+      console.log('element exists')
+    }
```

This is a limitation of the current spec. There is a `registerElement` but not `unregisterElement`. :astonished: 

Some solutions I've found workable:
- Wrap the `registerElement` call in a `try`/`catch` to swallow errors. Can also do this in a helper.
- Do the `registerElement` calls up front, like the bootstrapping in the routes or app source files.
- (For unit tests) Generate random element names each time you register the element to avoid left overs.